code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/AndroidUIService.java

```diff
@@ -241,9 +242,11 @@ public void showLocalNotification(final NotificationSetting notificationSetting)
         intent.putExtra(NOTIFICATION_REQUEST_CODE_KEY, requestCode);
         intent.putExtra(NOTIFICATION_DEEPLINK_KEY, notificationSetting.getDeeplink());
         intent.putExtra(NOTIFICATION_CONTENT_KEY, notificationSetting.getContent());
-        intent.putExtra(
-                NOTIFICATION_USER_INFO_KEY,
-                (HashMap<String, Object>) notificationSetting.getUserInfo());
+        final HashMap<String, Object> userInfo =
+                !MapUtils.isNullOrEmpty(notificationSetting.getUserInfo())
+                        ? new HashMap<>(notificationSetting.getUserInfo())
+                        : null;
+        intent.putExtra(NOTIFICATION_USER_INFO_KEY, userInfo);
```

Comment:
nit: you can do away with fetching UserInfo once by 
```
final HashMap<String, Object> userInfo = notificationSetting.getUserInfo();
intent.putExtra(NOTIFICATION_USER_INFO_KEY, (userInfo != null ? new HashMap<>(userInfo) : null));
```

Further, I think `NOTIFICATION_USER_INFO_KEY` can only be added if contents exist instead of adding a null value explicitly.
```
    final HashMap<String, Object> userInfo = notificationSetting.getUserInfo();
    if (!MapUtils.isNullOrEmpty(userInfo)) {
        intent.putExtra(NOTIFICATION_USER_INFO_KEY, new HashMap<>(userInfo));
    }
```

Reply:
`notificationSetting.getUserInfo()` is returning a Map so assigning it to `final HashMap<String, Object> userInfo` will not work without casting (which what the original problematic code was doing). I'll update the new code to only add the NOTIFICATION_USER_INFO_KEY if the user info map is non empty.